app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/stats/VpnServiceStateLogger.kt

```diff
@@ -60,7 +60,20 @@ class VpnServiceStateLogger @Inject constructor(
     private val dispatcher = Executors.newSingleThreadExecutor().asCoroutineDispatcher()
     private val job = ConflatedJob()
 
+    override fun onVpnStarting(coroutineScope: CoroutineScope) {
+        job += coroutineScope.launch(dispatcher) {
+            logcat { "VpnServiceStateLogger, new state ENABLING" }
+            vpnDatabase.vpnServiceStateDao().insert(VpnServiceStateStats(state = VpnServiceState.ENABLING))
+        }
```

Comment:
Synced with @aitorvs and we agreed that this change will fix the race condition I mentioned above:
```suggestion
        logcat { "VpnServiceStateLogger, new state ENABLING" }
        vpnDatabase.vpnServiceStateDao().insert(VpnServiceStateStats(state = VpnServiceState.ENABLING))
```

This way, we don't switch threads when starting the VPN.